app/controllers/events_controller.rb

```diff
@@ -9,13 +9,29 @@ def all
   end
 
   def create
-
     originating_data = { message: params[:message] }
-    steps = JSON.parse(request.body.read, symbolize_names: true)
+    steps_encoded = request.body.read
+    steps = JSON.parse(steps_encoded, symbolize_names: true)
 
-    workflow = Workflow.build(steps: steps)
+    sequence = Sequence.new
+    sequence.name = steps[0][:name]
+    sequence.steps = steps_encoded
+    sequence.save
 
-    result = workflow.start(originating_data)
+    render plain: 'ok'
+  end
+
+  def fire
+    sequence = Sequence.where(name: params[:message]).first
+
+    return render plain: "no sequence found for \"#{params[:message]}\"" unless sequence
+
+    steps = JSON.parse(sequence.steps, symbolize_names: true)
+    steps.each { |s| s[:config] = { name: s[:name] } }
+    steps.unshift({ type: 'WorkflowInitiator' })
```

Comment:
Here we grab a sequence from the DB, add a beginning "web initiator event" and then make a workflow from that and fire the whole thing.